go/arrow/array/dictionary.go

```diff
@@ -672,7 +672,7 @@ func NewDictionaryBuilderWithDict(mem memory.Allocator, dt *arrow.DictionaryType
 }
 
 func NewDictionaryBuilder(mem memory.Allocator, dt *arrow.DictionaryType) DictionaryBuilder {
-	return NewDictionaryBuilderWithDict(mem, dt, nil)
+	return NewDictionaryBuilderWithDict(mem, dt, dt.Values)
```

Comment:
@zeroshade I've left my safe world of util/protobuf_reflect here.
For handling the enums, where we know all the values in advance, we can buildWithDict. But the array needed to be somewhere. I think here was the least disruptive place to put it.

Reply:
This isn't a viable solution unfortunately. The dictionary values should not be tied to the dictionary data type at all. You need to explicitly handle holding onto the values internally to the protobuf reflect code and call `NewDictionaryBuilderWithDict` yourself from there. We can't propagate the values in the `DictionaryType` struct

Reply:
that's fair enough. At the moment. I have a recordBuilder from `array.NewRecordBuilder()` and I pass `recordBuilder.Field(i)` when I appendValueOrNull. 

I can't replace the builder there with a new `NewDictionaryBuilderWithDict`, do you have any advice on how to do that bit?

Reply:
Those `InsertDictValues` methods are on types like `array.BinaryDictionaryBuilder` which all take the private `dictionaryBuilder` as an input.

Do you prefer making `dictionaryBuilder` public, or having a 2nd public one, similar to the `builder` `Builder`, `dictionaryBuilder`, `DictionaryBuilder` pattern?

Reply:
if you have a `RecordBuilder` then you would already have the dictionary builder existing. In this scenario with protobuf, since we're talking about enums, aren't we always going to have a `StringDictionaryBuilder` so you could just do a type assertion?

Reply:
unfortunately not since: `array.BinaryDictionaryBuilder` does not implement `array.DictionaryBuilder`
missing Type and the Append methods

Reply:
updated, 👍 thanks for the help